test/ILLink.RoslynAnalyzer.Tests/DynamicallyAccessedMembersAnalyzerTests.cs

```diff
@@ -19,6 +21,568 @@ static Task VerifyDynamicallyAccessedMembersAnalyzer (string source, params Diag
 				expected: expected);
 		}
 
+		static Task VerifyDynamicallyAccessedMembersCodeFix (
+			string source,
+			string fixedSource,
+			DiagnosticResult[] baselineExpected,
+			DiagnosticResult[] fixedExpected,
+			int? numberOfIterations = null)
+		{
+			var test = new VerifyCS.Test {
+				TestCode = source,
+				FixedCode = fixedSource,
+				ReferenceAssemblies = TestCaseUtils.Net6PreviewAssemblies
+			};
+			test.ExpectedDiagnostics.AddRange (baselineExpected);
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{MSBuildPropertyOptionNames.EnableTrimAnalyzer} = true")));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+#region CodeFixTests
+		[Fact]
+		public async Task DiagID_2070 ()
+		{
+			var test = @"
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+class C
+{
+	public static void Main()
+	{
+		M(typeof(C));
+	}
+	static void M(Type T)
+	{
+		T.GetMethods();
+	}
+}";
+			var fixtest = @"
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+class C
+{
+	public static void Main()
+	{
+		M(typeof(C));
+	}
+	static void M([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)] Type T)
+	{
+		T.GetMethods();
+	}
+}";
+			await VerifyDynamicallyAccessedMembersCodeFix (test, fixtest, new[] {
+				// /0/Test0.cs(17,6): warning IL2070: 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicMethods' in call to 'System.Type.GetMethods()'. The parameter 'T' of method 'C.M(Type)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.
+				VerifyCS.Diagnostic(DiagnosticId.DynamicallyAccessedMembersMismatchParameterTargetsThisParameter).WithSpan(13, 3, 13, 17).WithArguments("System.Type.GetMethods()", "T", "C.M(Type)", "'DynamicallyAccessedMemberTypes.PublicMethods'") },
+				fixedExpected: Array.Empty<DiagnosticResult> ());
+		}
+
+				[Fact]
+		public async Task DiagID_2070NonPublic ()
+		{
+			var test = @"
+using System;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+
+class C
+{
+	public static void Main()
+	{
+		M(typeof(C));
+	}
+	static void M(Type T)
+	{
+		T.GetMethods(BindingFlags.NonPublic);
+	}
+}";
+			var fixtest = @"
+using System;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+
+class C
+{
+	public static void Main()
+	{
+		M(typeof(C));
+	}
+	static void M([DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.NonPublicMethods)] Type T)
```

Comment:
PR is failing because the code fixer adds `DynamicallyAccessedMemberTypes.PublicMethods | DynamicallyAccessedMemberTypes.NonPublicMethods`